server/monitor/src/main/java/org/apache/accumulo/monitor/util/ParameterValidator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.monitor.util;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Simple utility class to validate Accumulo Monitor Query and Path parameters
+ */
+public class ParameterValidator {
+
+  /**
+   * @param s
+   *          String parameter to sanitize. Common usages are tableId, type, tserverAddress
+   * @return URLEncoder encoded version of the passed in String parameter
+   * @throws UnsupportedEncodingException
+   *           if the string cannot be encoded to UTF-8
+   */
+  public static String sanitizeParameter(String s) throws UnsupportedEncodingException {
+    return StringUtils.isEmpty(s) ? StringUtils.EMPTY : URLEncoder.encode(s, "UTF-8").trim();
```

Comment:
I don't think this should be returning a URL-encoded version of the string. In most cases, we actually need to use the value after it has been decoded by Jersey. The only time we'd want to re-encode it is before display... but this method doesn't seem to be limited to only being used prior to display in the view.

It should also not throw an UnsupportedEncodingException. UTF-8 is always supported. This exception should be suppressed inside the method, instead of leaking out.

Also, `StandardCharsets.UTF_8.name()` should be used instead of `"UTF-8"`, whenever the String version of the charset is required.